libraries/ms-drivers/src/stm32f0xx/mcp23008_gpio_expander.c

```diff
@@ -0,0 +1,84 @@
+#include "mcp23008_gpio_expander.h"
+
+#include <stdbool.h>
+#include "i2c.h"
+#include "mcp23008_gpio_expander_defs.h"
+
+StatusCode mcp23008_gpio_init(const Mcp23008I2CAddress i2c_address) {
```

Comment:
don't think it's necessary, have the user initialize i2c. We may use i2c with other stuff as well, in that case this module makes it incompatible. In addition, you can't initialize i2c three times if you want to use this module three times.